### ISSUE TITLE
Script capturing input from stdin

### ISSUE THREAD

**Issue by tavmem** [kevinlawler/kona#351]
This k feature (missing from Kona) was identified by Bakul.
In k2.8:
```
tom@xps ~/ktest $ ls
k  kscript.k
tom@xps ~/ktest $ cat kscript.k
#! ./k
x:0:`"/dev/fd/0"
`0:x
tom@xps ~/ktest $ ls | ./kscript.k
k
kscript.k

tom@xps ~/ktest $ 
```
In Kona:
```
tom@xps ~/konaTst $ ls
k  kscript.k
tom@xps ~/konaTst $ cat kscript.k
#! ./k
x:0:`"/dev/fd/0"
`0:x
tom@xps ~/konaTst $ ls | ./kscript.k
Invalid argument error
>  
```

**Comment by tavmem** [comment]
Detail some of the differences encounted with the second line of kscript.k:
In k2.8-Linux, the above will work whether or not the backtick is included.
Without backtick:
```
tom@xps ~/ktest $ ls
k  kscript.k
tom@xps ~/ktest $ cat kscript.k
#! ./k
x:0:"/dev/fd/0"
`0:x
tom@xps ~/ktest $ ls | ./kscript.k
k
kscript.k
```
In Kona, the above fails with the backtick included (as shown above).
Although it works without the backtick, only the first item is returned from "ls".
Without backtick:
```
tom@xps ~/konaTst $ ls
k  kscript.k
tom@xps ~/konaTst $ cat kscript.k
#! ./k
x:0:"/dev/fd/0"
`0:x
tom@xps ~/konaTst $ ls | ./kscript.k
k
```
There is another significant difference:
Kona waits for a single line of input from stdin, then returns a single string:
```
  0:"/dev/fd/0"
abc
"abc"
```
k2.8-Linux waits only about 1 second.  If no input arrives, it returns an empty list.
```
  0:`"/dev/fd/0"
()
```
If you are really quick, you can key in some data, and get it back.
```
  0:"/dev/fd/0"            // key in "a", "Enter", "s"
a
s
(,"a"
 ,"s")

  0:"/dev/fd/0"           // key in "a" with no "Enter"
a
,,"a"
```
In summary, there are at least 3 (and maybe 4) items that need to be addressed:
1) The command should work with or without the backtick
2) It should accept multiple lines of data
3) The result should be a list of strings
4) k2.8-Linux only waits about 1 second. Kona waits till input arrives.

**Comment by tavmem** [comment]
Currently in Kona, the following 3 commands are handled exactly the same:
```
  0:"/dev/fd/0"
  0:"/dev/fd/stdin"
  0:`
```
In k2.8-Linux, each command is  handled differently:
The second one does not work at all:
```
  0:"/dev/fd/stdin"
/dev/fd/stdin: No such file or directory
0:"/dev/fd/stdin"
^
> 
```
In k2.8-Linux:
The first command handles multiple lines of input, which need to be ready immediately.
The third command waits for input, but then only processes one line of input.

BTW: The first command addresses the concern raised by @kirbyfan64 in issue #350 that there be a way to read all lines of input.

**Comment by bakul** [comment]
On Thu, 11 Jun 2015 10:59:41 PDT Tom Szczesny <notifications@github.com> wrote:
> 
> Currently in Kona, the following 3 commands are handled exactly the same:
> ```
>   0:"/dev/fd/0"
>   0:"/dev/fd/stdin"

There is /dev/fd/0 and there is /dev/stdin but no /dev/fd/stdin.
The first two are the same.  Kona shouldn't look at the name
but accept what the system returns on open for read.

>   0:`

This should be handled like k3 (that is, k-2.x or k-3.x).

> ```
> In k2.8-Linux, each command is  handled differently:
> The second one does not work at all:
> ```
>   0:"/dev/fd/stdin"
> /dev/fd/stdin: No such file or directory

Correct. This name never existed.

> In k2.8-Linux:
> The first command handles multiple lines of input, which need to be ready imm
> ediately.

I don't know about k-2.8 but I have k-2.95 and k-3.2 and they
both behave as follows:

The input doesn't have to be ready immediately. k will
continue reading until the input terminates. Using the below 
sample program x.k,

    #! env k
    x:0:`"/dev/fd/0"
    `0:x

the input can be from a file, as in

    ./x.k <some-file

or a pipe, as in

    ls|./x.k

or a fifo, as in

    mkfifo foofi
    ./x.k < foofi & date > foofi

If you just do 

    ./x.k

k3 is smart enough to let you type anything you want until you
hit ^D, at this point it doesn't terminate as in previous
cases but it will allow you to type k code on the console.

Note that k3 seems to be tolerant about filenames as either a
string like "foo" or a symbol like `"foo".

> The third command waits for input, but then only processes one line of input.

Correct.

IMHO kona should behave the same way.

Kona should start an interative session only if 'isatty(0)'
is true. Otherwise it should terminate after reading the
input.

The only thing kona has to special case is the behavior of 0:`
and `0: verbs. Basically just use open() for any name given
except a solitary `.

**Comment by tavmem** [comment]
Thanks !
